index.js

```diff
@@ -76,7 +76,9 @@ class StylusCompiler {
     if (Array.isArray(plugins)) {
       const handler = plugin => compiler.use(plugin());
       plugins.forEach(pluginName => {
-        if (Array.isArray(pluginName)) {
+        if (Object.prototype.toString.call(pluginName) === '[object Function]') {
```

Comment:
I feel you. I have a strong distaste for `typeof` because it has burned me in the past, but I made the requested change.